packages/@react-spectrum/actiongroup/src/ActionGroup.tsx

```diff
@@ -108,39 +234,155 @@ function ActionGroupItem<T>({item, state, isDisabled, isEmphasized, onAction}: A
     });
   }
 
+  // If button text is hidden, we need to show it as a tooltip instead, so
+  // go find the text element in the DOM after rendering.
+  let textId = useId();
+  let [textContent, setTextContent] = useState('');
+  useLayoutEffect(() => {
+    if (hideButtonText) {
+      setTextContent(document.getElementById(textId)?.textContent);
+    }
+  }, [hideButtonText, item.rendered, textId]);
+
   let button = (
     // Use a PressResponder to send DOM props through.
     // ActionButton doesn't allow overriding the role by default.
     <PressResponder {...mergeProps(buttonProps, hoverProps)}>
-      <ActionButton
-        ref={ref}
-        UNSAFE_className={
-          classNames(
-            styles,
-            'spectrum-ActionGroup-item',
-            {
-              'is-selected': isSelected,
-              'is-hovered': isHovered
-            },
+      <SlotProvider
```

Comment:
I'm surprised this works, there's a SlotProvider inside of the button already https://github.com/adobe/react-spectrum/blob/main/packages/%40react-spectrum/button/src/ActionButton.tsx#L64 that should be clearing this one?

Reply:
Looks like it merges them: https://github.com/adobe/react-spectrum/blob/a4831e5fba17cf0a740c4c601e0c298d99960de3/packages/%40react-spectrum/utils/src/Slots.tsx#L39